datastream-server-api/src/main/java/com/linkedin/datastream/server/FlushlessEventProducerHandler.java

```diff
@@ -36,15 +31,24 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(FlushlessEventProducerHandler.class);
 
+  private final String _offsetCheckpointTrackingStrategy;
   private final DatastreamEventProducer _eventProducer;
-  private final ConcurrentHashMap<SourcePartition, CallbackStatus> _callbackStatusMap = new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<SourcePartition, CallbackStatus<T>> _callbackStatusMap = new ConcurrentHashMap<>();
 
   /**
    * Constructor for FlushlessEventProducerHandler
    */
-  public FlushlessEventProducerHandler(DatastreamEventProducer eventProducer) {
+  public FlushlessEventProducerHandler(DatastreamEventProducer eventProducer, String offsetCheckpointTrackingStrategy) {
     _eventProducer = eventProducer;
     _eventProducer.enablePeriodicFlushOnSend(false);
+    _offsetCheckpointTrackingStrategy = offsetCheckpointTrackingStrategy;
+  }
+
+  /**
+   * Creating a new instance of the OffsetCheckpointTrackingStrategy to be used for checkpointing
+   */
+  public CallbackStatus<T> createInstanceOffsetCheckpointTrackingStrategy() {
```

Comment:
This method name is misleading and I am not sure how to infer the name. From what I have understood, You want to create CallbackStatus object. This method can be simplified to `createCallbackStatusObject`. 

The same name issue applies to offsetCheckpointTrackingStrategy, where you are keeping CallbackStatus class.